knowledge-base/window-keep-content-when-closed.md

```diff
@@ -0,0 +1,222 @@
+
+How to preserve the Window content upon closing?
+
+Is it possible to close/hide the Window with CSS?
+
+## Solution
+
+To achieve the desired result you can toggle the visibility of the Window with CSS. For that purpose:
+
+1. Create a [custom "Close" action]({%slug components/window/actions%}#custom-actions).
+1. Handle the [`OnClick`]({%slug button-events%}#onclick) event of the action button to apply the needed CSS.
+
+The further steps will vary depending on whether or not the [Window is modal]({%slug components/window/modal%}). See details and examples below:
+* [Custom close for non-modal Window](#custom-close-for-non-modal-Window)
+* [Custom close for modal Window](#custom-close-for-modal-Window)
+
+### Custom close for non-modal Window
+
+The next steps for custom closing a non-modal Window are:
+
+3. In the `OnClick` handler of the custom close action, set a custom CSS class to the Window through the `Class` parmeter.
+
+4. Use this class to add `display: none;` rule to the Window.
+
+5. Clear this class when the user opens the Window to revert the custom style and display the componet.
+
+````CSHTML
+@*Hide Window with CSS to preserve its content*@
+
+@using System.ComponentModel.DataAnnotations
+
+<style>
+    .hidden-window {
+        display: none;
+    }
+</style>
+
+<TelerikWindow @bind-Visible="@WindowRendered"
+               Class="@WindowClass">
+    <WindowTitle>
+        Window Title
+    </WindowTitle>
+    <WindowContent>
+
+        <TelerikForm Model="@person">
+            <FormValidation>
+                <DataAnnotationsValidator />
+            </FormValidation>
+        </TelerikForm>
+
+    </WindowContent>
+    <WindowActions>
+        <WindowAction Name="CustomClose"
+                      Icon="@FontIcon.X"
+                      OnClick="@CustomCloseHandler" />
+    </WindowActions>
+</TelerikWindow>
+
+<TelerikButton OnClick="@OpenWindowHandler">Open Window</TelerikButton>
+
+@code {
+    private bool WindowRendered { get; set; }
+
+    private string WindowClass { get; set; }
+
+    private void CustomCloseHandler()
+    {
+        WindowClass = "hidden-window";
+    }
+
+    private void OpenWindowHandler()
+    {
+        if (!WindowRendered)
+        {
+            WindowRendered = true;
+        }
+
+        WindowClass = "";
+    }
+
+    private Person person = new Person();
+
+    public class Person
+    {
+        [Required]
+        public int Id { get; set; }
+
+        [Required]
+        [Display(Name = "First Name")]
+        public string FirstName { get; set; }
+
+        [Required]
+        [Display(Name = "Last Name")]
+        public string LastName { get; set; }
+
+        [Required]
+        [Display(Name = "Date of Birth")]
+        public DateTime? DOB { get; set; }
+    }
+}
+````
+
+### Custom close for modal Window
```

Comment:
You may want to mention the Dialog component somewhere. Just to clarify which part of the article is relevant to it.